ci/scripts/utils/launch_java_agent.sh

```diff
@@ -56,22 +108,49 @@ cd "${JENKINS_AGENT_LANUCH_DIR}"
 
 if ! [[ -f agent.jar ]]; then
   curl -sO "${controller_url}/jnlpJars/agent.jar"
+  echo "Updated agent.jar downloaded"
 fi
 
-JENKINS_TOKEN=$(cat jenkins_token)
+if [[ ! -f "${controller_user_auth_token}" ]]; then
+   echo "User Jenkins authetication TOKEN to the controller for using the Remote API does not exist"
+   exit 1
+fi
+
+JENKINS_TOKEN=$(cat "${controller_user_auth_token}")
+echo -e "#!/usr/bin/env python
+import json,sys
+with open(sys.argv[1], 'r') as file:
+    data = json.load(file)
+print(data.get('offline','True'))
+" > parse.py
+chmod u+x parse.py
+
+check_node_online() {
+    rm -f curl_response
+    curl_response=$(curl --silent -u "${controller_user}:${JENKINS_TOKEN}" "${controller_url}/computer/${MACHINE_ID^}-EMC/api/json?pretty=true") || true
+    echo -n "${curl_response}" > curl_response
+    ./parse.py curl_response
+}
 
-# 
-offline=$(curl --silent -u "${controller_user}:${JENKINS_TOKEN}" "${controller_url}/computer/${MACHINE_ID^}-EMC/api/json?pretty=true" | grep '\"offline\"' | awk '{gsub(/,/,"");print $3}') || true
-echo "Jenkins Agent offline setting: ${offline}"
+offline=$(set -e; check_node_online)
 
-if [[ "${offline}" == "true" ]]; then
-  echo "Jenkins Agent is offline. Lanuching Jenkins Agent on ${host}"
-  command="nohup ${JAVA} -jar agent.jar -jnlpUrl ${controller_url}/computer/${MACHINE_ID^}-EMC/jenkins-agent.jnlp  -secret @jenkins-secret-file -workDir ${JENKINS_WORK_DIR}"
-  echo -e "Lanuching Jenkins Agent on ${host} with the command:\n${command}" >& "${LOG}"
-  ${command} >> "${LOG}" 2>&1 &
-  nohup_PID=$!
-  echo "Java agent running on PID: ${nohup_PID}" >> "${LOG}" 2>&1
-  echo "Java agent running on PID: ${nohup_PID}"
+if [[ "${offline}" != "False" ]]; then
+  if [[ "${1:-}" != "-f" ]]; then
+      echo "Jenkins Agent is offline. Waiting 5 more minutes to check again in the event it is a temp network issue"
+      sleep 300
+  fi
```

Comment:
Sorry I wasn't clear.  What I meant here was that `now` (which has been changed presently to `-f`), does not force the Jenkins agent to start.  It only prevents sleeping for 300 seconds.  To force the Jenkins agent to start, I think you will need another if block.

What I was trying to suggest was to keep `now` as `now` and **add** a `-f` flag to force the Jenkins agent to start.

Reply:
It should check first, that is the intention of the switch "do not double check" just check once.  Please do not change that intention.  Thanks.